components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtDBQueries.java

```diff
@@ -94,8 +94,10 @@ public class ApplicationMgtDBQueries {
             "TENANT_ID = ? AND APP_NAME LIKE ? AND APP_NAME != ? ORDER BY ID DESC";
     public static final String LOAD_APP_COUNT_BY_TENANT = "SELECT COUNT(*) FROM SP_APP WHERE TENANT_ID = ? AND " +
             "APP_NAME != ? ";
-    public static final String LOAD_APP_COUNT_BY_TENANT_AND_APP_NAME = "SELECT COUNT(*) FROM SP_APP WHERE " +
-            "TENANT_ID = ? AND APP_NAME LIKE ? AND APP_NAME != ? ";
+
+    public static final String LOAD_APP_COUNT_BY_TENANT_AND_APP_NAME = "SELECT COUNT(SP_APP.APP_NAME) FROM SP_APP " +
+            "INNER JOIN SP_INBOUND_AUTH ON SP_APP.ID = SP_INBOUND_AUTH.APP_ID WHERE SP_APP.TENANT_ID = ? " +
+            "AND (%s) AND SP_APP.APP_NAME != ? ";
```

Comment:
What do we expect to have here? Won't this cause SQL injection issues?

Reply:
@pamodaaw %s is replaced by the filter starting generated and sent by the API. This won't be vulnerable to SQL injection since this string is generated in the API side through a binary tree which is created by validated filter params input by the user.

Reply:
Fixed the vulnerability by using prepared statement approach.